main.py

```diff
@@ -1 +1,179 @@
+    """Returns a version of the garden where a day has passed"""
+    
+
+    # return garden
+    
+    for row in range(horizontal):
+        for col in range(vertical):
+            item = garden[row][col]
+            plant(garden, row - 1, col , item)
+            plant(garden, row + 1, col , item)
+            plant(garden, row, col - 1 , item)
+            plant(garden ,row, col + 1,item)
+            
+
+    
+    return garden
+
+def grow_garden(garden , horizontal , vertical):
+    days = int(input("How many days do you want to simulate? "))
+
+    for _ in range(days):
+        garden = day_increment(
+            garden, horizontal, vertical
+        )  # Update garden with the latest state
+        
+        print("\n----------\n")
+        
+        display_garden(garden, horizontal, vertical)
+
+def input_coordi(horizontal , vertical):
+    x = int(input('Enter the row number'))
+    y = int(input("Enter the column numbers"))
+    while x > horizontal or y > vertical  or x < 0 or y < 0 :
+        print("either your row or column number is invalid")
+        x = int(input('Enter the row number'))
+        y = int(input("Enter the column numbers"))
+    
+    return x-1 , y-1
+
+
+
+   
+
+if __name__ == "__main__":
+    horizontal = int(input("Enter the number of rows in the garden: "))
+    vertical = int(input("Enter the number of columns in the garden: "))
+
+    # call make and display garden functions
+    garden = create_garden(horizontal, vertical)
+    display_garden(garden, horizontal, vertical)
+    cont = True
+    while cont:
+        # main code
+
+        # ask the user choice menu
+        choice = int(
+            input(
+                "What do you want to do? Plant,remove weeds, or count the number of plants and weeds. Enter 1, 2, 3: press 4 for passing N days "
+            )
+        )
+
+        if choice == 1:
+
+            x, y = input_coordi(horizontal , vertical)
+        
+            if garden[x][y] != EMPTY_CELL:
+                print("Invalid, item already present in that area.")
+            else:
+                garden[x][y] = FLOWER
+            display_garden(garden, horizontal, vertical)
+
+        elif choice == 2:
+            x, y = input_coordi(horizontal , vertical)
+            if garden[x][y] != WEED:
+                print("Invalid, no weed in the area.")
+            else:
+                garden[x][y] = EMPTY_CELL
+
+            display_garden(garden, horizontal, vertical)
+
+        elif choice == 4:
+            grow_garden(garden, horizontal, vertical)
+
+        elif choice == 3:
+            flower_num, weed_num = counting_items(garden, horizontal, vertical)
+            print(f"The number of flowers are {flower_num}")
+            print(f"The number of weeds are {weed_num}")
+
+        continue_ = str(input("Do you want to continue? y or n: ")).lower()
+        if continue_ == "y":
+            cont = True
+            continue
+        else:
+            cont = False
```

Comment:
Following this alternative approach would yield the same behavior:
```suggestion
        if continue_ != "y":
            cont = False
```